src/Compilers/CSharp/Portable/Syntax/SyntaxNodeExtensions.cs

```diff
@@ -67,6 +67,9 @@ internal static bool CanHaveAssociatedLocalBinder(this CSharpSyntaxNode syntax)
                 kind == SyntaxKind.CatchFilterClause ||
                 kind == SyntaxKind.SwitchSection ||
                 kind == SyntaxKind.MatchSection ||
+                kind == SyntaxKind.EqualsValueClause ||
```

Comment:
EqualsValueClause has a scope now, but it has nothing to do with FieldInitializerBinders. The binder is created by the LocalBinderFactory as the comment says.